boot/bootutil/src/boot_record.c

```diff
@@ -330,6 +330,12 @@ int boot_save_shared_data(const struct image_header *hdr, const struct flash_are
                                           BLINFO_RUNNING_SLOT,
                                           sizeof(slot), (void *)&slot);
     }
+
+    if (!rc) {
+        rc = boot_add_data_to_shared_area(TLV_MAJOR_BLINFO,
+                                          BLINFO_FLASH_AREA,
+                                          sizeof(fap->fa_id), (void *)&fap->fa_id);
+    }
```

Comment:
Three issues:
 1) i am trying to disentangle MCUboot from Flash Map API as this entanglement blocks us from freely modifying both Zephyr and MCUboot, and so the usage of flash area id and flash_area within MCUboot is going to be phased out. There is also problem with flash area functions having different prototypes and flash_area objects having different members, depending on a system.
2) area id in MCUboot and Zephyr app that got boot does not have to be the same; it is enough if there is storage before images that gets removed with overlay in mcuboot and numbering will differ
3) mcuboot should not directly access flash_area objects, rather use getter functions like flash_area_get_id.

Reply:
Ok, makes sense. I have no issue changing this, really all I want is a workable solution for indicating the flash area that Zephyr's MCUBoot subsystem should use when marking an image as confirmed. Is there an API for mapping the boot slot number back to a flash area we can use?